```
  test//test_string_conversion_stream_operator.cc:107:66: error: passing no argument for the '...' parameter of a variadic macro is a C++20 extension [-Werror,-Wc++20-extensions]
    107 | TYPED_TEST_SUITE(WideFixedToStringConversionTest, TestTargetTypes);
        |                                                                  ^
  /usr/local/include/gtest/gtest-typed-test.h:191:9: note: macro 'TYPED_TEST_SUITE' defined here
    191 | #define TYPED_TEST_SUITE(CaseName, Types, ...)                          \
        |         ^
  1 error generated.
```

--- a/Makefile
+++ b/Makefile
@@ -14,9 +14,11 @@ INCLUDE_DIR_HEADER += $(wildcard $(INCLUDE_DIR)/*.h)
 dummy := $(shell type $(CXX))  # In GNU Make, default CXX should be g++
 ifeq ($(.SHELLSTATUS),0)
   CXX := clang++
+  MY_CXX_MAX_OPTIMIZE_FLAG := -Ofast
   dummy := $(shell type $(CXX))
   ifeq ($(.SHELLSTATUS),0)
     CXX := c++
+    MY_CXX_MAX_OPTIMIZE_FLAG := -O3 -ffast-math
   endif
 endif
 
@@ -65,7 +67,7 @@ ifeq ($(BUILD_TYPE), release)
 ifneq ($(findstring coverage,$(MAKECMDGOALS)),)
 $(error Use BUILD_TYPE=coverage for target "coverage")
 endif
-BUILD_TYPE_CXXFLAGS += -Ofast
+BUILD_TYPE_CXXFLAGS += $(MY_CXX_MAX_OPTIMIZE_FLAG)
 BUILD_TYPE_CXXFLAGS += -DNDEBUG
 else ifeq ($(BUILD_TYPE), debug)
 ifneq ($(findstring coverage,$(MAKECMDGOALS)),)

--- a/Makefile
+++ b/Makefile
@@ -119,7 +119,7 @@ WARNING_CXXFLAGS += -Wno-error=inline
 
 # Build C++ compiler flags for test.
 TEST_CXXFLAGS := $(CXXFLAGS)
-TEST_CXXFLAGS += --std=c++17
+TEST_CXXFLAGS += --std=c++20
 TEST_CXXFLAGS += $(addprefix -I, $(INCLUDE_DIR))
 TEST_CXXFLAGS += $(addprefix -I, $(SYSTEM_INCLUDE_DIRS))
 TEST_CXXFLAGS += $(BUILD_TYPE_CXXFLAGS)

